File: examples/ruby/websocket/autobahn_client.rb
require 'rubygems'
require File.expand_path('../../../../lib/faye', __FILE__)
require 'cgi'

EM.run {
  host   = 'ws://localhost:9001'
  agent  = "Faye (Ruby #{RUBY_VERSION})"
  cases  = 0
  skip   = [247,248,249]
  
  socket = Faye::WebSocket::Client.new("#{host}/getCaseCount")
  
  socket.onmessage = lambda do |event|
    puts "Total cases to run: #{event.data}"
    cases = event.data.to_i
  end
  
  socket.onclose = lambda do |event|
    run_case = lambda do |n|
      if n > cases
        socket = Faye::WebSocket::Client.new("#{host}/updateReports?agent=%22#{CGI.escape agent}%22")
        socket.onclose = lambda { |e| EM.stop }
        
      elsif skip.include?(n)
        EM.next_tick { run_case.call(n+1) }
        
      else
        puts "Running test case ##{n} ..."
        socket = Faye::WebSocket::Client.new("#{host}/runCase?case=#{n}&agent=%22#{CGI.escape agent}%22")
        
        socket.onmessage = lambda do |event|
          socket.send(event.data)
        end
        
        socket.onclose = lambda do |event|
          run_case.call(n + 1)
        end
      end
    end
    
    run_case.call(1)
  end
}

File: examples/node/websocket/autobahn_client.js
var faye   = require('../../../build/faye-node');

var host  = 'ws://localhost:9001',
    agent = 'Faye (Node ' + process.version + ')',
    cases = 0,
    skip  = [247,248,249];

var socket = new faye.WebSocket.Client(host + '/getCaseCount');

socket.onmessage = function(event) {
  console.log('Total cases to run: ' + event.data);
  cases = parseInt(event.data);
};

socket.onclose = function() {
  var runCase = function(n) {
    if (n > cases) {
      socket = new faye.WebSocket.Client(host + '/updateReports?agent=' + encodeURIComponent(agent));
      socket.onclose = process.exit
      
    } else if (skip.indexOf(n) >= 0) {
      runCase(n + 1);
      
    } else {
      console.log('Running test case #' + n + ' ...');
      socket = new faye.WebSocket.Client(host + '/runCase?case=' + n + '&agent=' + encodeURIComponent(agent));
      
      socket.onmessage = function(event) {
        socket.send(event.data);
      };
      
      socket.onclose = function() {
        runCase(n + 1);
      };
    }
  };
  
  runCase(1);
};

